replaced QGIS project templates

severin-prenitzer opened the pull request:
replaced pylovo_db_geom.qgz by template_local_db.qgz and template_remote_db.qgz. Both new templates are based on the old one, but have slightly adjusted styles, dynamic styling with project variables and dynamic version filtering with a project variable and virtual layers instead of layer filters. template_remote_db.qgz additionally has a dynamic data source by using a service definition (makes changing the data source simpler but needs an additional config file; this one is for the use with QWC).

bereba commented on QGIS/template_local_db.qgz:
transformer_positions and lines_result are not properly connected when I open it locally. Do you have the same issues?

bereba commented on QGIS/template_local_db.qgz:
Transformer numbers with 3 digits are not within circles (e.g. 1.24)

bereba commented on QGIS/template_local_db.qgz:
The coordinate system of lines is per default wrong (3035) and has to be changed manually to 4326.
Can we also configure this?

bereba commented:
You could per default select some layers. I suggest the following:
- raw_data/postcode
- raw_data/transformers
- grids/buildings_result
- grids/transformer_positions
- grids/lines_result

severin-prenitzer commented:
1) transformer_positions and lines_result are connected correctly for me. If you use an old database schema you might experience issues because those two are virtual layers with more complicated queries. I needed these queries because of Matyas' changes to the database schema. Version filtering for most layers works like that (no additional table needed):

SELECT *
FROM buildings_result_embedded
WHERE version_id = var('version_id')

But for lines_result we now need a nested query with grid_result, because the lines_result table does not contain version_id itself anymore:

SELECT * 
FROM lines_result_embedded 
WHERE grid_result_id IN (
	SELECT grid_result_id
	FROM grid_result_embedded
	WHERE version_id = var('version_id')
);

If your database uses the old schema, this query might not work.
(for transformer_positions the problem is similar)
![Screenshot (24)](https://github.com/user-attachments/assets/8c02cea1-2a42-4e37-9f6b-0d5cce8ec985)

severin-prenitzer commented:
2) I adjusted circle and text size for the transformers, now 3-digit numbers are inside the circles
3) I changed the coordinate system for the lines_result layer, it should now by default be correct
4) The size of the raw_data/transformers and grid/transformer_position circles matches for me. I think it only differs when re-rendering while zooming in or out
5) I set your suggested default layer selection

(all changes applied to both template files)

severin-prenitzer commented:
Regarding the config files, i guess the pg_service.conf (the file that contains the connection details which is referenced by every layer) would make sense, because one cannot use the remote template without it. But i would also add a short README file that explains what the pg_service.conf does and how to edit the project variables for version filtering and style changes.
